### PR TITLE
IE-3551: Add new app service plan instance types (PxV3) and autoscaling rule

### PR DESCRIPTION
**Pull request by lp-code**

### CODE REVIEW

File: resources/resourceAppServicePlanAutoscaleSetting/azuredeploy.jsonc [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    },
+    "resources": [
+      {
+        "type": "Microsoft.Insights/autoscalesettings",
+        "name": "[variables('settingName')]",
+        "apiVersion": "2015-04-01",
+        "location": "[parameters('location')]",
+        "tags": "[parameters('tags')]",
+        "properties": {
+          "profiles": [
```
Comment (jlyshoel) [inline resources/resourceAppServicePlanAutoscaleSetting/azuredeploy.jsonc:95]: I'm wondering if this array (profiles) actually should be moved/specified outside, and just a single input (which begs the question if we need it as a shared template)

The reason is that I think we should account for multiple metrics, i.e. scaling both with CPU, memory and network load (since most operations in moe are CPU heavy).

File: resources/resourceAppServicePlanAutoscaleSetting/azuredeploy.jsonc [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+                    "operator": "GreaterThan",
+                    "threshold": "[parameters('metricThresholdToScaleOut')]",
+                    "dividePerInstance": false
+
+                  },
+                  "scaleAction": {
+                    "direction": "Increase",
+                    "type": "ChangeCount",
+                    "value": "[parameters('changeCountScaleOut')]",
+                    "cooldown": "PT5M"
```
Comment (jlyshoel) [inline resources/resourceAppServicePlanAutoscaleSetting/azuredeploy.jsonc:122]: Also have some input regarding cooldown and how aggressively we scale up vs. down, but we can do that discussion in another PR if we decide to move the "profiles" array outside of shared :-)

File: resources/resourceAppServicePlan/azuredeploy.jsonc [hunk 3]
```diff
@@ -12,6 +12,9 @@
         "P1V2", // Premium V2 Small
         "P2V2", // Premium V2 Medium
         "P3V2", // Premium V2 Large
+        "P1V3", // Premium V3 Small
+        "P2V3", // Premium V3 Medium
+        "P3V3", // Premium V3 Large
```
Comment (jlyshoel) [inline resources/resourceAppServicePlan/azuredeploy.jsonc:17]: 👍

**Comment by lp-code** [comment]
Moved the scaling rules definition to the application template. New (quite short now) review required, @jlyshoel.

**Review by jlyshoel — APPROVED** [review]
